res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,438 @@
+    }
+};
+
+// The "loupe" button that enables/disables track scrolling
+Mixage.scrollToggle = function(channel, control, value, _status, _group) {
+    // calculate deck number from MIDI control. 0x03 controls deck 1, 0x12 deck 2
+    var deckNr = control === 0x03 ? 1 : 2;
+    // check for pressed->release or released->press
+    if (Mixage.scrollToggleLastState[deckNr] !== value) {
+        Mixage.scrollToggleLastState[deckNr] = value;
+        if (value === 0) {
+            return;
+        }
+        Mixage.setScrolling(deckNr, !Mixage.isScrollActive[deckNr]);
+    }
+};
+
+// The touch function on the wheels that enables/disables scratching
+Mixage.wheelTouch = function(channel, control, value, _status, _group) {
+    // calculate deck number from MIDI control. 0x24 controls deck 1, 0x25 deck 2
+    var deckNr = control === 0x24 ? 1 : 2;
+    // check if scratching should be enabled
+    if (Mixage.scratchByWheelTouch || Mixage.isScratchActive[deckNr]) {
+        if (value === 0x7F) {
+            // turn on scratching on this deck
+            var alpha = 1.0 / 8.0;
+            var beta = alpha / 32.0;
+            engine.scratchEnable(deckNr, Mixage.scratchTicksPerRevolution, 33.0 + 1.0 / 3.0, alpha, beta);
+        } else {
+            engine.scratchDisable(deckNr);
+        }
+    }
+};
+
+// The wheel that controls the scratching / jogging
+Mixage.wheelTurn = function(channel, control, value, _status, _group) {
+    // calculate deck number from MIDI control. 0x24 controls deck 1, 0x25 deck 2
+    var deckNr = control === 0x24 ? 1 : 2;
+    // only enable wheel if functionality has been enabled
+    if (Mixage.scratchByWheelTouch || Mixage.isScratchActive[deckNr] || Mixage.isScrollActive[deckNr]) {
+        // control centers on 0x40 (64), calculate difference to that value
+        var newValue = value - 64;
+        if (Mixage.isScrollActive[deckNr]) { // scroll deck
+            var controlString = "[Channel" + deckNr + "]";
+            var keyString = "playposition";// + newValue > 0 ? "up_small" : "down_small";
+            var currentPosition = engine.getValue(controlString, keyString);
+            var speedFactor = 0.00005;
+            engine.setValue(controlString, keyString, currentPosition + speedFactor * newValue * Mixage.jogWheelScrollSpeed);
+        } else if (engine.isScratching(deckNr)) {
+            engine.scratchTick(deckNr, newValue); // scratch deck
+        } else {
+            engine.setValue("[Channel" + deckNr + "]", "jog", newValue); // pitch bend deck
+        }
+    }
+};
+
+// The MASTER button that toggles a deck as sync leader / master
+Mixage.handleDeckSyncMode = function(_channel, _control, _value, _status, _group) {
+    // Function currently not working as of 2.3.3: https://manual.mixxx.org/2.4/gl/chapters/appendix/mixxx_controls.html#control-[ChannelN]-sync_master
+    // Disable until this is working
```

Comment:
can you elaborate?

Reply:
> Changed in version 2.3.0: This button just enables [sync lock](https://manual.mixxx.org/2.4/gl/glossary.html#term-sync-lock) mode (similar to [[ChannelN],sync_enabled](https://manual.mixxx.org/2.4/gl/chapters/appendix/mixxx_controls.html#control-[ChannelN]-sync_enabled)), it does not actually guarantee the deck will be the sync leader. This will be fixed in a future version.

What seems to happen when enabling `sync_master` on a deck is it just enables synching on all decks, but either deck follows the other? I've tried setting `sync_mode`, but that did not work either. So I disabled that function altogether.

Reply:
Yes, thats expected. The feature of sync leader is only apparent when you have tracks with variable BPM (and also analyzed / created the beatmap as such). The leader deck will then also drive BPMs of the followers.

Reply:
Unfortunately the feature is not yet complete and also undocumented.

Reply:
So, is it ok if we leave it like that? Or should I remove the mapping in the XML and the function completely?
Regular sync lock can still be enabled by holding the sync button.

Reply:
Yeah, leave it like that.